### PR TITLE
Refactor read

### PR DESCRIPTION
**Pull request by reneesun**
* refactored `read a promotion`.
* removed 404 error handler. After I refactored the code, it didn't call this error handler any more.
* changed to api.url_for in `add a new promotion`
* changed the 404 error message in `read a promotion` to pass the step of `Then I should see the message "404 Not Found" ` in build. I will change the message back after the BDD file is changed.

### CODE REVIEW

File: service/service.py [hunk 1]
```diff
@@ -132,6 +131,26 @@ class PromotionResource(Resource):
...
+        """
+        Retrieve a single Promotion
+
+        This endpoint will return a Promotion based on it's id
+        """
+        app.logger.info("Request to Retrieve a promotion with id [%s]", promotion_id)
+        promotion = Promotion.find(promotion_id)
+        if not promotion:
+            api.abort(status.HTTP_404_NOT_FOUND, 
+                      "404 Not Found")
```
Comment (iamnwi) [inline service/service.py]: Maybe you can change this message to `"404 Not Found: Promotion with id '{}' was not found.".format(promotion_id))` to include more information.

Comment (reneesun) [reply service/service.py]: Sure. It's better to include more information.

**Comment by reneesun** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/a903ceab-af58-4d40-9de5-39501ad15e1a?env_id=ibm%3Ayp%3Aus-south "nyu-devops-fall19-promotion-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/5860bddd-fc68-4c80-afbc-55ea22efe279 "nyu-devops-fall19-promotion-pipeline") deployed [promotions](https://github.com/devops-fall-2019-promotions-squad/promotions/releases/tag/deploy-ibm-yp-us-south-rs6565-nyu.edu-dev-20191209-003910 "deploy-ibm-yp-us-south-rs6565-nyu.edu-dev-20191209-003910") to [dev](https://console.bluemix.net/apps/efcd7d31-ed8b-433d-b811-007435b814df?env_id=ibm:yp:us-south "ibm:yp:us-south:rs6565@nyu.edu:dev"), including commit ded03cae657fc7e5a8fc4424557b7583e4d77549

**Comment by reneesun** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/a903ceab-af58-4d40-9de5-39501ad15e1a?env_id=ibm%3Ayp%3Aus-south "nyu-devops-fall19-promotion-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/5860bddd-fc68-4c80-afbc-55ea22efe279 "nyu-devops-fall19-promotion-pipeline") deployed [promotions](https://github.com/devops-fall-2019-promotions-squad/promotions/releases/tag/deploy-ibm-yp-us-south-rs6565-nyu.edu-prod-20191209-004350 "deploy-ibm-yp-us-south-rs6565-nyu.edu-prod-20191209-004350") to [prod](https://console.bluemix.net/apps/ce7c172b-4a93-4d4a-be2f-bbf7bd1fb4b7?env_id=ibm:yp:us-south "ibm:yp:us-south:rs6565@nyu.edu:prod"), including commit ded03cae657fc7e5a8fc4424557b7583e4d77549

**Pull request merged**